File: rb101_rb109/easy9/how_long.rb
=begin
input: a string
output: return an array contains every word from the string, to which you have appended a space and the word length.
rules: assume that words in the string are separated by exactly one space, and that any substring of non-space characters is a word
=end

def word_lengths(str)
  arr = str.split
  return arr if arr.size == 0
  arr.map do |sub_str|
    sub_str << " #{sub_str.size}"
  end  
end


# another solution
=begin
def word_lengths(string)
  string.split.map { |word| "#{word} #{word.length}" }
end
=end

long_str = "cow sheep chicken"
p word_lengths(long_str)

#p word_lengths("baseball hot dogs and apple pie")

#p word_lengths("It ain't easy, is it?")

#p word_lengths("Supercalifragilisticexpialidocious")

#p word_lengths("") == []

p long_str

File: ls215/js210_string_processing/how_long.js
/*
Question: a function that returns an aray with each word and the count of letters in the word
  - arg. a string of words separated bby single space
  - return an array of strings ex. ['apple 5', 'cat 3']
algo:
  - create an empty array
  - split string by words
  - count letters of each word
  - add the word and count to array
  - return array
*/

function wordLengths(string) {
  let result = [];

  if (string === undefined || string === '') {
    return result; 
  }

  string.split(' ').forEach(word => {
    result.push(word.concat(' ', String(word.length)));
  })

  return result;
}

console.log(wordLengths('cow sheep chicken'));
console.log(wordLengths('baseball hot dogs and apple pie'));
console.log(wordLengths("It ain't easy, is it?"));
console.log(wordLengths('Supercalifragilisticexpialidocious'));
console.log(wordLengths(''));
console.log(wordLengths());
